File: swig/lua/test-r_bin.lua
-- RBin --
require "r_bin"

file = arg[1] or "/bin/ls"
b = r_bin.RBin ()
b:load (file, "")
baddr = b:get_baddr ()
s = b:get_sections ()
nsects = s:size() - 1
for i=0,nsects do
	print (string.format ('offset=0x%08x va=0x%08x size=%05i %s',
				s[i].offset, baddr+s[i].rva, s[i].size, s[i].name))
end

-- Introspection
--m = getmetatable(s)
--table.foreach(m, print)
--table.foreach (m['.fn'], print)

File: swig/go/test-r_bin.go
package main

import (
	"os"
	"fmt"
	"r_bin"
	)

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage:", os.Args[0], "<bin>")
		os.Exit(1)
	}
	b := r_bin.NewRBin()
	b.Load(os.Args[1], "")
	baddr := b.GetBaddr()
	fmt.Println("-> Sections")
	fmt.Printf("baddr=%08x\n", baddr)
	for _, s:= range b.Get_sections() {
		fmt.Printf("offset=0x%08x va=0x%08x size=%05d %s\n",
				s.GetOffset(), baddr+s.GetRva(), s.GetSize(), s.GetName())
	}
}
